### PR TITLE
Implement a simple LCD menu

### PR DESCRIPTION
**Pull request by numo68**
## Description:

The integration implements a hierarchical menu on character-based LCD displays, primarily meant to be used together with a rotary encoder with a single push button. It is possible to navigate the menu, changing enum values (named integer) and float ones, with the values themselves being maintained in globals (that allows to use them freely in other places, persisting them etc). Automations are added to perform user-defined functionality on entering or leaving the menu level and changing the value or triggering a command. The menu can be hidden and shown again, enabling alternating between the menu and other usage for the display.

**Pull request in [esphome](https://github.com/esphome/esphome) with YAML changes (if applicable):** esphome/esphome#3406

## Checklist:

  - [X] Branch: `next` is for changes and new documentation that will go public with the next ESPHome release. Fixes, changes and adjustments for the current release should be created against `current`.
  - [X] Link added in `/index.rst` when creating new documents for new components or cookbook.

### CODE REVIEW

File: components/display-menu.rst [hunk 1]
```diff
@@ -0,0 +1,791 @@
...
+          id(my_lcd_menu).draw();
+          if (!id(my_lcd_menu).is_active())
+            it.print("Menu is not active");
+
+    # Declare a LCD menu
+    lcd_menu:
+      id: my_lcd_menu
+      display_id: my_lcd
+      dimensions: 20x4
+      active: True
```
Comment (jesserockz) [inline components/display-menu.rst]: Please change all `True` and `False` to be `true` and `false`
```suggestion
      active: true
```

**Review by jesserockz — CHANGES_REQUESTED** [review]
This should start out as either `lcd_menu.rst` or split into `lcd_menu.rst` for specific `lcd_menu` stuff and a `display_menu.rst` for the common stuff. (Also underscore not hypen please in the filename)

File: components/display-menu.rst (hunk 1 above)
Comment (numo68) [reply components/display-menu.rst]: I have split the description to the common and `lcd_menu` specific part; I created a `display_menu` subdirectory as it is expected that a pixel-oriented display arrives here at some day. True/False changed to lowerspace (frankly the existing code has them half/half so it is hard to tell what is expected). Names have underscores. Thanks.

**Comment by nagyrobi** [comment]
Not stale

**Review by jesserockz — APPROVED** [review]

**Pull request merged**